### DISCUSSION TITLE
Not stitching all photos

### DISCUSSION THREAD

**Discussion by StraLopez** [OpenStitching/stitching#255] [Q&A]
Hi, this is my code. I want to create a 360 panorama with 8 photos. When I run the code, I get the following warning:

C:\Users\Utente\stich\Lib\site-packages\stitching\subsetter.py:32: StitchingWarning: Not all images are included in the final panorama. If this is not intended, use the 'matches_graph_dot_file' parameter to analyze your matches. You might want to lower the 'confidence_threshold' or try another 'detector' to include all your images.
  warnings.warn(

////////

there are the photos used:
![0](https://github.com/user-attachments/assets/df8bfdc4-91b3-4f12-9718-d120b424525b)
![1](https://github.com/user-attachments/assets/f8c43e24-1384-4d23-a44e-0b39ede46c55)
![2](https://github.com/user-attachments/assets/55e8540d-7218-462a-93bb-04894bf3d264)
![3](https://github.com/user-attachments/assets/ee3cfc20-5076-41f1-9e3d-d77b58590716)
![4](https://github.com/user-attachments/assets/4ede595f-f562-4789-9b55-d472a91684e5)
![5](https://github.com/user-attachments/assets/aa4095c1-504f-4e71-a04f-be63f7a6b9d9)
![6](https://github.com/user-attachments/assets/92a8c390-257f-4b5f-8f49-9525a438efee)
![7](https://github.com/user-attachments/assets/58c45f93-bc21-428b-8a3d-cf536e5f0c37)

The result is a panorama made up of only two photos, instead of all eight.

![panorama_result](https://github.com/user-attachments/assets/adcae30e-e694-472a-acad-4d8b78977a05)

  
  
import cv2 as cv
from stitching import Stitcher
import os

settings = {"detector": "sift", "confidence_threshold": 0.00001, "matches_graph_dot_file": "matches_graph.dot"}
stitcher = Stitcher(**settings)

folder_path = r"C:\Users\Utente\Downloads\fuori2"

image_filenames = [
    os.path.join(folder_path, f"{i}.png") for i in range(0, 8)
]

panorama = stitcher.stitch(image_filenames)

output_path = os.path.join(folder_path, "panorama_result.jpg")

cv.imwrite(output_path, panorama)

cv.imshow("Panorama", panorama)
cv.waitKey(0)
cv.destroyAllWindows()

I tryed with (sift, orb), i tryed also to modity the trhershold (0.1, 0.00001, 0.5) nothing changed

**Comment by lukasalexanderweber** [comment]
You need more overlap. The last two Images don't have an overlapping area at all

**Reply by lukasalexanderweber** [reply]
Also the first and the last

**Reply by StraLopez** [reply]
so i need more foto and closer for overlap?

**Reply by StraLopez** [reply]
an other little question...this script work also with video?

**Reply by lukasalexanderweber** [reply]
Yes: https://github.com/OpenStitching/stitching/discussions/148

**Reply by StraLopez** [reply]
wow...amazing work!
